src/editors/editDocumentCodeLensProvider.ts

```diff
@@ -64,7 +69,7 @@ implements vscode.CodeLensProvider {
         // to be able to save the editable document.
         if (item !== null && item._id && namespace) {
           codeLensesInfo.push({
-            documentId: item._id,
+            documentId: EJSON.deserialize(EJSON.serialize(item._id)),
```

Comment:
I deserialize ids here and not in `MongoDBDocumentService` to keep `fetchDocument` clean from any data mutation and make it the controllers' responsibility to prepare data in the required format.